internal/cli/generate_tiltfile.go

```diff
@@ -0,0 +1,175 @@
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/fatih/color"
+	"golang.org/x/term"
+)
+
+type generateTiltfileResult string
+
+const (
+	generateTiltfileError          = "error"
+	generateTiltfileUserExit       = "exit"
+	generateTiltfileNonInteractive = "non_interactive"
+	generateTiltfileAlreadyExists  = "already_exists"
+	generateTiltfileCreated        = "created"
+	generateTiltfileDeclined       = "declined"
+)
+
+// maybeGenerateTiltfile offers to create a Tiltfile if one does not exist and
+// the user is running in an interactive (TTY) session.
+//
+// The generateTiltfileResult value is ALWAYS set, even on error.
+func maybeGenerateTiltfile(tfPath string) (generateTiltfileResult, error) {
+	if !term.IsTerminal(int(os.Stdin.Fd())) || !term.IsTerminal(int(os.Stdout.Fd())) {
+		return generateTiltfileNonInteractive, nil
+	}
+
+	if absPath, err := filepath.Abs(tfPath); err != nil {
+		// the absolute path is really just to improve CLI output, if the path
+		// itself is so invalid this fails, we'll catch + report it via the
+		// logic to determine if a Tiltfile already exists
+		tfPath = absPath
+	}
+
+	if hasTiltfile, err := checkTiltfileExists(tfPath); err != nil {
+		// either Tiltfile path is totally invalid or there's something like
+		// a permissions error, so report it & exit
+		return generateTiltfileError, err
+	} else if hasTiltfile {
+		// Tiltfile exists, so don't prompt to generate one
+		return generateTiltfileAlreadyExists, nil
+	}
+
+	t, restoreTerm, err := setupTerm(os.Stdin)
+	if err != nil {
+		return generateTiltfileError, nil
+	}
+
+	lineCount := 0
+	var postFinishMessage string
+	defer func() {
+		_ = restoreTerm()
+
+		if postFinishMessage != "" {
+			// NOTE: pre-win10, there's no support for ANSI escape codes, and
```

Comment:
😂

Reply:
Ha! I meant to add some more color to the PR description about some of my logic/decisions but I'll give a quick TL;DR here:
* The low-level term lib we already have is `tcell` - this basically gives you a character grid, so not going near that directly for this feature 🙅 
* Tilt has its own internal package called `rty` (built on top of `tcell`) which is what powers the current TUI - it's obviously very powerful, and I did give it a shot, but was hitting some issues, and since we're planning for this to be temporary and extreme interactivity/in-place updates weren't critical, it didn't seem worth spending more time on
* I really didn't want to pull in another dependency for this (same logic as in previous point)

Visually, it's not quite as spiffy in Windows as a result, but it still behaves totally reasonably IMO, so this was a compromise I felt okay with. Otherwise, I definitely wouldn't have been going near raw ANSI escape codes 🙀